File: BOJ/11723. 집합/solution.cc
// https://www.acmicpc.net/problem/11723
#include <iostream>
#include <string>

using namespace std;

int main() {
    ios_base::sync_with_stdio(0);
    cin.tie(0);

    int len, n, res = 0;
    string fn;

    cin >> len;
    for (int i = 0; i < len; i += 1) {
        cin >> fn;
        if (fn[0] == 'a'&& fn[1] == 'd') {
            cin >> n;
            res |= (1 << n);
        }
        else if (fn[0] == 'r') {
            cin >> n;
            res &= ~(1 << n);
        }
        else if (fn[0] == 'c') {
            cin >> n;
            res & (1 << n) ? cout << "1\n" : cout << "0\n";
        }
        else if (fn[0] == 't') {
            cin >> n;
            res & (1 << n) ? res &= ~(1 << n) : res |= (1 << n);
        }
        else if (fn[1] == 'l') res = (1 << 21) - 1;
        else res = 0;
    }

    return 0;
} 

File: BOJ/11723. 집합/solution.js
// https://www.acmicpc.net/problem/11723
const query = require('fs').readFileSync('/dev/stdin').toString().split('\n');
const len = +query[0];
let bit = 0;

for (let i = 1; i <= len; i += 1) {
    const [fn, x] = query[i];

    if (fn === 'add') bit |= (1 << x);
    else if (fn === 'remove') bit &= -1 * ~(1 << x);
    else if (fn === 'check') console.log(`${bit & (1 << x) ? 1 : 0}`);
    else if (fn === 'toggle') bit & (1 << x) ? bit &= -1 * ~(1 << x) : bit |= (1 << x);
    else if (fn === 'all') bit = (1 << 21) - 1;
    else bit = 0;
}